lib/types/result-set.js

```diff
@@ -10,30 +10,76 @@ const asyncIteratorSymbol = Symbol.asyncIterator || "@@asyncIterator";
 
 /** @module types */
 
-// Old definition of the Result set
-// /**
-//  * Creates a new instance of ResultSet.
-//  * @class
-//  * @classdesc Represents the result of a query.
-//  * @param {Object} response
-//  * @param {String} host
-//  * @param {Object} triedHosts
-//  * @param {Number} speculativeExecutions
-//  * @param {Number} consistency
-//  * @param {Boolean} isSchemaInAgreement
-//  * @constructor
-//  */
-// function ResultSet(response, host, triedHosts, speculativeExecutions, consistency, isSchemaInAgreement) {
-
 /**
  * Represents the result of a query.
  */
 
 class ResultSet {
+    // Old definition of the Result set
+    // /**
+    //  * @param {Object} result
+    //  * @param {String} host
+    //  * @param {Object} triedHosts
+    //  * @param {Number} speculativeExecutions
+    //  * @param {Number} consistency
+    //  * @param {Boolean} isSchemaInAgreement
+    //  */
     /**
      * @param {rust.QueryResultWrapper} result
      */
-    constructor(result) {
+    constructor(
+        result,
+        host,
+        triedHosts,
+        speculativeExecutions,
+        consistency,
+        isSchemaInAgreement,
+    ) {
+        if (!(result instanceof rust.QueryResultWrapper)) {
+            console.warn(
+                "Using legacy version of the ResultSet constructor.\n" +
+                    "This is deprecated and may be removed at any moment.",
+            );
+            if (speculativeExecutions === -1) {
+                speculativeExecutions = 0;
+            }
```

Comment:
Hmm? This is _a bit_ suspicious...
Absolutely explain this in a comment.

Reply:
That's the old logic. It was commented with ``// if no execution was made at all, set to 0.`` so I can add this back.